sklearn/model_selection/tests/test_search.py

```diff
@@ -885,8 +885,16 @@ def test_param_sampler():
 def check_cv_results_array_types(search, param_keys, score_keys):
     # Check if the search `cv_results`'s array are of correct types
     cv_results = search.cv_results_
-    assert all(isinstance(cv_results[param], np.ma.MaskedArray) for param in param_keys)
-    assert all(cv_results[key].dtype == object for key in param_keys)
+    assert all(
+        isinstance(cv_results[param], np.ma.MaskedArray)
+        for param in param_keys
+        if param not in {"iter", "n_resources"}
+    )
+    assert all(
+        cv_results[key].dtype == object
+        for key in param_keys
+        if key not in {"iter", "n_resources"}
+    )
```

Comment:
This modification is not need if we define ``iter`` and ``n_resources`` as extra keys as explained in my previous comment.

Reply:
I see, thank you.